Assets/Plugins/Source/Core/EOSOnConnectLogin.cs

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2021 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using Epic.OnlineServices;
+    using Epic.OnlineServices.Connect;
+    using System;
+    using UnityEngine;
+
+    /// <summary>
+    /// Contains common implementations for the IEOSOnConnectLogin interface.
+    /// Not all instances where IEOSOnConnectLogin is implemented are replaced
+    /// by this abstract class, because a gradual conversion will have a smaller
+    /// impact.
+    /// </summary>
+    public abstract class EOSOnConnectLogin : IEOSOnConnectLogin
+    {
+        /// <summary>
+        /// The product user id associated with the connect login that called
+        /// the connect login callback for this manager.
+        /// </summary>
+        protected ProductUserId _productUserId;
```

Comment:
Hmmm... have we created an additional source of truth for the product user id? Should we pull from EOSManager / EOSSingleton?

Reply:
Maybe if all this function is doing is passing this along on a successful login, instead of storing the value here, it's just passed in as an argument to PostSuccessfulLogin?

Reply:
This one is a little confusing to me to be honest. 

The way `EOSAchievementsManager` _currently_ caches the data is by using a `ConcurrentDictionary` that maps a `ProductUserId` type key to a `List<Stats>` type value. What's odd (to me) is that implies that more than one `ProductUserId` might be present - in which case storing the `ProductUserId` instance for the _manager_ might make sense. Like - this would store the `ProductUserId` that is _active_ when the manager is "logged in."

I'm a huge fan of reducing the number of sources of truth to be a singularity - almost to a fault. 

So, clarity on this would probably be good to get. @andrew-hirata-playeveryware Do you have insight you can lend?

Reply:
The short answer is that we shouldn't store the _productUserID in this class as the EOS Manager should be the 'owner' of the product user ID.